### PR TITLE
MCSC-1018 Add Existing LTA Screens to Triage Flow

### PR DESCRIPTION
**Pull request by bharti5verma**

### CODE REVIEW

File: app/controllers/setupquestions/lifetimeallowance/ChangeInLifetimeAllowanceController.scala [hunk 1]
```diff
@@ -60,13 +63,17 @@ class ChangeInLifetimeAllowanceController @Inject() (
...
+
             for {
-              updatedAnswers <- Future.fromTry(request.userAnswers.set(ChangeInLifetimeAllowancePage, value))
-              redirectUrl     = ChangeInLifetimeAllowancePage.navigate(mode, updatedAnswers).url
-              answersWithNav  = LTASection.saveNavigation(updatedAnswers, redirectUrl)
-              _              <- userDataService.set(answersWithNav)
+              updatedAnswers          <- Future.fromTry(request.userAnswers.set(ChangeInLifetimeAllowancePage, value))
+              updatedAnswersWithStatus = LTAKickOutStatus().saveLTAKickOutStatus(updatedAnswers, ltaKickOutStatus)
+              redirectUrl              = ChangeInLifetimeAllowancePage.navigate(mode, updatedAnswersWithStatus).url
+              answersWithNav           = LTASection.saveNavigation(updatedAnswersWithStatus, redirectUrl)
```
Comment (MichaelAAbbott) [inline app/controllers/setupquestions/lifetimeallowance/ChangeInLifetimeAllowanceController.scala:73]: This should be setup section I believe.

File: app/controllers/setupquestions/lifetimeallowance/NewLTAChargeController.scala [hunk 2]
```diff
@@ -60,13 +63,17 @@ class ChangeInTaxChargeController @Inject() (
...
+
             for {
-              updatedAnswers <- Future.fromTry(request.userAnswers.set(ChangeInTaxChargePage, value))
-              redirectUrl     = ChangeInTaxChargePage.navigate(mode, updatedAnswers).url
-              answersWithNav  = LTASection.saveNavigation(updatedAnswers, redirectUrl)
-              _              <- userDataService.set(answersWithNav)
+              updatedAnswers          <- Future.fromTry(request.userAnswers.set(NewLTAChargePage, value))
+              updatedAnswersWithStatus = LTAKickOutStatus().saveLTAKickOutStatus(updatedAnswers, ltaKickOutStatus)
+              redirectUrl              = NewLTAChargePage.navigate(mode, updatedAnswersWithStatus).url
+              answersWithNav           = LTASection.saveNavigation(updatedAnswersWithStatus, redirectUrl)
```
Comment (MichaelAAbbott) [inline app/controllers/setupquestions/lifetimeallowance/NewLTAChargeController.scala:73]: This should be setup section I believe.

File: app/controllers/setupquestions/lifetimeallowance/OtherSchemeNotificationController.scala [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        .fold(
+          formWithErrors => Future.successful(BadRequest(view(formWithErrors, mode))),
+          value => {
+            val ltaKickOutStatus = if (value) kickOutStatusCompleted else kickOutStatusTrue
+
+            for {
+              updatedAnswers          <- Future.fromTry(request.userAnswers.set(OtherSchemeNotificationPage, value))
+              updatedAnswersWithStatus = LTAKickOutStatus().saveLTAKickOutStatus(updatedAnswers, ltaKickOutStatus)
+              redirectUrl              = OtherSchemeNotificationPage.navigate(mode, updatedAnswersWithStatus).url
+              answersWithNav           = LTASection.saveNavigation(updatedAnswersWithStatus, redirectUrl)
```
Comment (MichaelAAbbott) [inline app/controllers/setupquestions/lifetimeallowance/OtherSchemeNotificationController.scala:73]: This should be setup section I believe.

File: app/controllers/setupquestions/lifetimeallowance/PreviousLTAChargeController.scala [hunk 4]
```diff
@@ -64,7 +64,7 @@ class PreviousLTAChargeController @Inject() (
             for {
               updatedAnswers <- Future.fromTry(request.userAnswers.set(PreviousLTAChargePage, value))
               redirectUrl     = PreviousLTAChargePage.navigate(mode, updatedAnswers).url
-              answersWithNav  = SetupSection.saveNavigation(updatedAnswers, redirectUrl)
+              answersWithNav  = LTASection.saveNavigation(updatedAnswers, redirectUrl)
```
Comment (MichaelAAbbott) [inline app/controllers/setupquestions/lifetimeallowance/PreviousLTAChargeController.scala:67]: This should be setup section I believe.

File: app/views/setupquestions/lifetimeallowance/IncreaseInLTAChargeView.scala.html [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        @govukRadios(
+            RadiosViewModel.yesNo(
+                field = form("value"),
+                legend = LegendViewModel(messages("increaseInLTACharge.heading")).asPageHeading()
+            )
+        )
+
+        @govukButton(
+            ButtonViewModel(messages("site.continue"))
+        )
```
Comment (MichaelAAbbott) [inline app/views/setupquestions/lifetimeallowance/IncreaseInLTAChargeView.scala.html:44]: Needs save and return button

File: app/views/setupquestions/lifetimeallowance/NewLTAChargeView.scala.html [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        @govukRadios(
+            RadiosViewModel.yesNo(
+                field = form("value"),
+                legend = LegendViewModel(messages("newLTACharge.heading")).asPageHeading()
+            )
+        )
+
+        @govukButton(
+            ButtonViewModel(messages("site.continue"))
+        )
```
Comment (MichaelAAbbott) [inline app/views/setupquestions/lifetimeallowance/NewLTAChargeView.scala.html:44]: Needs save and return button

File: app/views/setupquestions/lifetimeallowance/OtherSchemeNotificationView.scala.html [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+        @govukRadios(
+            RadiosViewModel.yesNo(
+                field = form("value"),
+                legend = LegendViewModel(messages("otherSchemeNotification.heading")).asPageHeading()
+            )
+        )
+
+        @govukButton(
+            ButtonViewModel(messages("site.continue"))
+        )
```
Comment (MichaelAAbbott) [inline app/views/setupquestions/lifetimeallowance/OtherSchemeNotificationView.scala.html:44]: Needs save and return button

File: test/controllers/setupquestions/lifetimeallowance/ChangeInLifetimeAllowanceControllerSpec.scala [hunk 8]
```diff
@@ -79,7 +79,7 @@ class ChangeInLifetimeAllowanceControllerSpec extends SpecBase with MockitoSugar
       }
     }
 
-    "must redirect to the ChangeInTaxCharge page when valid data is submitted" in {
+    "must redirect to the ChangeInTaxCharge page when true is submitted" in {
```
Comment (MichaelAAbbott) [inline test/controllers/setupquestions/lifetimeallowance/ChangeInLifetimeAllowanceControllerSpec.scala:82]: Don't we just need to check for the 303 status here?

Comment (MichaelAAbbott) [reply test/controllers/setupquestions/lifetimeallowance/ChangeInLifetimeAllowanceControllerSpec.scala:82]: This applies to other controller specs as well.

File: test/controllers/setupquestions/lifetimeallowance/ChangeInLifetimeAllowanceControllerSpec.scala [hunk 9]
```diff
@@ -99,6 +99,31 @@ class ChangeInLifetimeAllowanceControllerSpec extends SpecBase with MockitoSugar
 
         val result = route(application, request).value
 
+        status(result) mustEqual SEE_OTHER
+        // redirectLocation(result).value mustEqual controllers.setupquestions.routes.CheckYourSetupAnswersController.onPageLoad().url
```
Comment (MichaelAAbbott) [inline test/controllers/setupquestions/lifetimeallowance/ChangeInLifetimeAllowanceControllerSpec.scala:103]: This can be removed.

**Review by MichaelAAbbott — CHANGES_REQUESTED** [review]
Controller specs also need a unit test to check the values for LTAKickOutStatus are setting correctly. See example here: https://github.com/hmrc/calculate-public-pension-adjustment-frontend/blob/5786ebd0601b491f274f520024e5569a33cdc647/test/controllers/setupquestions/annualallowance/NetIncomeAbove190KControllerSpec.scala#L157-L251

File: test/controllers/setupquestions/lifetimeallowance/ChangeInLifetimeAllowanceControllerSpec.scala (hunk 9 above)
Comment (bharti5verma) [reply test/controllers/setupquestions/lifetimeallowance/ChangeInLifetimeAllowanceControllerSpec.scala:103]: Done

File: test/controllers/setupquestions/lifetimeallowance/ChangeInLifetimeAllowanceControllerSpec.scala (hunk 8 above)
Comment (bharti5verma) [reply test/controllers/setupquestions/lifetimeallowance/ChangeInLifetimeAllowanceControllerSpec.scala]: Done

File: app/views/setupquestions/lifetimeallowance/OtherSchemeNotificationView.scala.html (hunk 7 above)
Comment (bharti5verma) [reply app/views/setupquestions/lifetimeallowance/OtherSchemeNotificationView.scala.html]: Done

File: app/views/setupquestions/lifetimeallowance/NewLTAChargeView.scala.html (hunk 6 above)
Comment (bharti5verma) [reply app/views/setupquestions/lifetimeallowance/NewLTAChargeView.scala.html]: Done

File: app/views/setupquestions/lifetimeallowance/IncreaseInLTAChargeView.scala.html (hunk 5 above)
Comment (bharti5verma) [reply app/views/setupquestions/lifetimeallowance/IncreaseInLTAChargeView.scala.html]: Done

File: app/controllers/setupquestions/lifetimeallowance/PreviousLTAChargeController.scala (hunk 4 above)
Comment (bharti5verma) [reply app/controllers/setupquestions/lifetimeallowance/PreviousLTAChargeController.scala]: Done

File: app/controllers/setupquestions/lifetimeallowance/OtherSchemeNotificationController.scala (hunk 3 above)
Comment (bharti5verma) [reply app/controllers/setupquestions/lifetimeallowance/OtherSchemeNotificationController.scala]: Done

File: app/controllers/setupquestions/lifetimeallowance/NewLTAChargeController.scala (hunk 2 above)
Comment (bharti5verma) [reply app/controllers/setupquestions/lifetimeallowance/NewLTAChargeController.scala]: Done

File: app/controllers/setupquestions/lifetimeallowance/ChangeInLifetimeAllowanceController.scala (hunk 1 above)
Comment (bharti5verma) [reply app/controllers/setupquestions/lifetimeallowance/ChangeInLifetimeAllowanceController.scala]: Done

**Review by MichaelAAbbott — APPROVED** [review]

**Review by MichaelAAbbott — APPROVED** [review]

**Pull request merged**